### DISCUSSION TITLE
Unexpected text trim

### DISCUSSION THREAD

**Discussion by Ggwppino** [GrapesJS/grapesjs#5520] [Q&A]
Hi, 

I have a problem once I select a portion of text and change its style from the editor, 3 textnodes was created in the project.
When I save the project by getting the data with `editor.getProjectData()`, it returns the text with 3 textnodes pre-applying (I guess) a trim.
So the complete sentence will have words attached to each other when reload the project with command `editor.loadProjectData(project)`.
I would like to know if there is something I'm missing, maybe some flag in the configuration to change.

What I expect should be the output when I reload the project
![Screenshot 2023-11-15 183325](https://github.com/GrapesJS/grapesjs/assets/29524875/87fefefc-6221-4886-a99f-4eb8c35fa6d8)

What I get as output when I reload the project
![Screenshot 2023-11-15 183353](https://github.com/GrapesJS/grapesjs/assets/29524875/fe6a2f14-8fb6-4ef3-99c9-20ae38733b69)

**Comment by artf** [comment]
Share a reproducible demo, please

**Reply by Ggwppino** [reply]
Actually these days I investigated better and there was nothing in grapesJS that did the trim. The trim was caused by a poorly set middleware on the server side that was trimming all the values ​​in the requests body. My bad 😅
